File: backend/boutique_app/src/main/kotlin/at/fhtw/dtos/Customer.kt
package at.fhtw.dtos

import org.springframework.format.annotation.DateTimeFormat
import java.sql.Date


data class Customer(
    val firstName: String,
    val lastName: String,
    val email: String,
    val phoneNumber: String,
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) val birthDate: Date
) {
    companion object {
        fun from(old: at.fhtw.model.entities.Customer): Customer {
            return Customer(
                old.firstName, old.lastName, old.email, old.phoneNumber, old.birthDate
            )
        }
    }

    fun toEntity(): at.fhtw.model.entities.Customer {
        return at.fhtw.model.entities.Customer(
            firstName, lastName, email, phoneNumber, birthDate
        )
    }
}

File: frontend/boutique_app/src/models/customer.ts
export default class Customer {
    firstName: string;
    lastName: string;
    email: string;
    phoneNumber: string;
    birthDate: string;
    address: string;

    constructor(firstName: string, lastName: string, email: string, phoneNumber: string, birthDate: string, address: string) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.birthDate = birthDate;
        this.address = address;
    }
}